### ISSUE TITLE
Why is there so much space between lines of my footnotes?

### ISSUE THREAD

**Issue by ctrlcctrlv** [sile-typesetter/sile#1027]
See e.g. footnote &numero;7 on page 6. 

https://raw.githubusercontent.com/ctrlcctrlv/PamudpodUnicodePaper/863487eca016bb938452c698705be329b5abe99e/pamudpod.pdf

**Comment by alerque** [comment]
The footnote frames are inheriting the linespace (leading) settings from either the default typesetter or perhaps your body one. They are clearly not appropriate for your smaller footnote font. You need to set the linespacing somehow. Since you're not using the book class I can't suggest exactly where, but even stuffing it in the footnote separator command would work. It could also be part of the footnote font command or baked into your footnote outputter if you've modified that.

Note the space between sequential insertions is determined differently using the insertion skip calculator.

**Comment by ctrlcctrlv** [comment]
@alerque For this document I am using the book class&mdash;how does that change things?

https://github.com/ctrlcctrlv/PamudpodUnicodePaper/blob/master/pamudpod.sil#L1

**Comment by ctrlcctrlv** [comment]
I looked at the code and if I'm understanding it right what it's doing seems objectively wrong.

## `packages/footnotes.lua`

```
  SILE.typesetter = oldT
  insertions.exports:insert("footnote", material)
```

So, changing things in the footnote frame typesetter won't do anything? This is why the size is only wrong inside a footnote and not between footnotes. :(

**Comment by alerque** [comment]
`\relax` man, that's not what I said at all. There is a good reason for the space between footnotes to be controlled separately (via `interInsertionSkip`) from regular line leading. I question whether it might not be better to re-use parskip for `interInsertionSkip` by default, but no it's not objectively wrong that baselineskip, parskip, and interinsertionskip are each their own thing.

> changing things in the footnote frame typesetter

Each insertion gets it's own typesetter. This is by design and quite useful sometimes.

Here is a fix for your leading:

```sile
\define[command=footnote:font]{\font[size=0.9em]{\set[parameter=document.baselineskip,value=0.5ex]\process\par}}
```

I'm re-opening this because we should probably make this more configurable via two commands instead of overloading one, maybe use a saner default (we hard code a smaller font size but leave the default leading), and document the result.

**Comment by Omikhleia** [comment]
I was about to open a new issue, but if i I understood correctly, this is the very topic of this one here.

That is (with a footnote:font size of 6pt in this example, for the sake of illustration):

![image](https://user-images.githubusercontent.com/18075640/132940895-4d33d2e1-5e80-4b9e-9a94-e1bf6e975f95.png)

The footnote text is displayed with the same baseline skip as the main text, so there is a lot of space between the lines (in long footnotes where line break applies).

I am not fond of the suggested fix above (setting an explicit baseline skip), because I have no idea what to put in there and it seems quite arbitrary (= test and try values until satisfied). The baseline skip is the default one (1.2em plus 1pt), relative to the font size, and one would have expected, I guess, that it would be applied to the footnote at its expected size, without any "trick". And that is kind of annoying ;)

Applying a quick and dirt hack to **footnotes** package as follows:

```
SILE.registerCommand("footnote", function (options, content)	SILE.registerCommand("footnote", function (options, content)
  SILE.call("footnotemark")					  SILE.call("footnotemark")
  local opts = SILE.scratch.insertions.classes.footnote		  local opts = SILE.scratch.insertions.classes.footnote
  local f = SILE.getFrame(opts["insertInto"].frame)		  local f = SILE.getFrame(opts["insertInto"].frame)
  local oldT = SILE.typesetter					  local oldT = SILE.typesetter
  SILE.typesetter = SILE.typesetter {}				  SILE.typesetter = SILE.typesetter {}
  SILE.typesetter:init(f)					  SILE.typesetter:init(f)
  SILE.typesetter.getTargetLength = function () return SILE.l	  SILE.typesetter.getTargetLength = function () return SILE.l
  SILE.settings.pushState()					  SILE.settings.pushState()
  SILE.settings.reset()						  SILE.settings.reset()
  local material = SILE.call("vbox", {}, function ()	      |	  local material
    SILE.call("footnote:font", {}, function ()		      |	  SILE.call("footnote:font", {}, function ()
							      >	      material = SILE.call("vbox", {}, function ()
      SILE.call("footnote:atstart", options)			      SILE.call("footnote:atstart", options)
      SILE.call("footnote:counter", options)			      SILE.call("footnote:counter", options)
      SILE.process(content)					      SILE.process(content)
    end)							    end)
  end)								  end)
  SILE.settings.popState()					  SILE.settings.popState()

```
I now get:

![image](https://user-images.githubusercontent.com/18075640/132940967-d0eca5c7-173e-4df7-ad03-970260d502d6.png)

This does seem far more logical (_or, at least, corresponding to what I would have expected_).

So to recap: if the font change is applied before constructing the `vbox`, the footnote lines seem more correct. 

Would that be a fix for the issue? The thing is, all those `xxx:font` hooks here and there are quite "open", one could put nearly anything in them (e.g. a `\set` as was suggested above, or say, some `\color`...), I am not sure what side effect it could have. But it seems to do the trick, no?
